legion_airflow/legion_airflow/hooks/k8s_base_hook.py

```diff
@@ -47,7 +48,10 @@ def _get_conn_from_k8s(cls, conn_id):
         :type conn_id: str
         """
         config_map = K8SSecretStorage.retrive(
-            storage_name=cls.STORAGE_NAME_PREFIX + conn_id,
+            storage_name=cls.STORAGE_NAME_PREFIX,
             k8s_namespace=os.environ['NAMESPACE']
         )
-        return Connection(**config_map.data)
+        connection_data = yaml.load(config_map.data['data'])
```

Comment:
Looks like you store all connections in key `data`. But I guess, it will be better to store each connection info in appropriate named key? And values of each info will be encoded via yaml